i2c/pcf8523_i2c/pcf8523_i2c.c

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "pico/stdlib.h"
+#include "pico/binary_info.h"
+#include "hardware/i2c.h"
+
+/* Example code to talk to a PCF8520 Real Time Clock module 
+
+   Connections on Raspberry Pi Pico board, other boards may vary.
+
+   GPIO PICO_DEFAULT_I2C_SDA_PIN (On Pico this is 4 (physical pin 6)) -> SDA on PCF8520 board
+   GPIO PICO_DEFAULT_I2C_SCK_PIN (On Pico this is 5 (physical pin 7)) -> SCL on PCF8520 board
+   5V (physical pin 40) -> VCC on PCF8520 board
+   GND (physical pin 38)  -> GND on PCF8520 board
+*/
+
+// By default these devices  are on bus address 0x68
+static int addr = 0x68;
+
+#ifdef i2c_default
+static void pcf8520_reset() {
+    // Two byte reset. First byte register, second byte data
+    // There are a load more options to set up the device in different ways that could be added here
+    uint8_t buf[] = {0x00, 0x58};
+    i2c_write_blocking(i2c_default, addr, buf, 2, false);
+}
+
+static void pcf820_write_current_time() {
+    // buf[0] is the register to write to
+    // buf[1] is the value that will be written to the register
+    uint8_t buf[2];
+
+    //Write values for the current time in the array
+    //index 0 -> second: bits 4-6 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 1 -> minute: bits 4-6 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 2 -> hour: bits 4-5 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 3 -> day of the month: bits 4-5 are responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 4 -> day of the week: where Sunday = 0x00, Monday = 0x01, Tuesday... ...Saturday = 0x06
+    //index 5 -> month: bit 4 is responsible for the ten's digit and bits 0-3 for the unit's digit
+    //index 6 -> year: bits 4-7 are responsible for the ten's digit and bits 0-3 for the unit's digit
+
+    //NOTE: if the value in the year register is a multiple for 4, it will be considered a leap year and hence will include the 29th of February
+
+    uint8_t current_val[7] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+    for (int i = 3; i < 10; ++i){
+        buf[0] = i;
+        buf[1] = current_val[i - 3];
```

Comment:
This looks a bit weird, and took me a while to figure out what it was doing. I think perhaps
```c
    for (int i = 0; i < 7; ++i) {
        buf[0] = i + 3;
        buf[1] = current_val[i];
```
might be _slightly_ more readable? :shrug: